remote-workstations/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -308,6 +314,31 @@ do
     fi
 done
 
+# Install and setup the Sumo Collector
+echo "-->Install SumoLogic collector"
+mkdir /tmp/sumo
+wget "https://collectors.sumologic.com/rest/download/linux/64" -O /tmp/sumo/SumoCollector.sh && sudo chmod +x /tmp/sumo/SumoCollector.sh
+wget "$ARTIFACT_LOCATION/user.properties$SAAS_TOKEN" -O /tmp/sumo/user.properties
+wget "$ARTIFACT_LOCATION/sumo-agent-vm-linux.json$SAAS_TOKEN" -O /tmp/sumo/sumo-agent-vm-linux.json
+JSON_FILE=/tmp/sumo/sumo-agent-vm-linux.json
+echo "Attemtping to set sumo collector ID to: " "$COLLECTOR_ID"
+sed -i s/collectorID/"$COLLECTOR_ID"/ /tmp/sumo/user.properties
+sed -i 's|syncsourceFile|'$JSON_FILE'|' /tmp/sumo/user.properties
+
+sudo /tmp/sumo/SumoCollector.sh -q -varfile user.properties
+
+# service collector install - configures the collector to start at boot time
+service collector install
+service collector restart
+service collector status
+
+# Install idle shutdown script
+echo "-->Install idle shutdown"
+mkdir /tmp/idleShutdown
+wget "$ARTIFACT_LOCATION/Install-Idle-Shutdown.sh$SAAS_TOKEN" -O /tmp/idleShutdown/Install-Idle-Shutdown-raw.sh 
+awk '{ sub("\r$", ""); print }' /tmp/idleShutdown/Install-Idle-Shutdown-raw.sh > /tmp/idleShutdown/Install-Idle-Shutdown.sh && sudo chmod +x /tmp/idleShutdown/Install-Idle-Shutdown.sh
+sudo /tmp/idleShutdown/Install-Idle-Shutdown.sh -install
```

Comment:
Here again -- are we assuming that this will always succeed? If it does fail, do we care or need to take any additional steps?

Reply:
We wrote the idle shutdown script, I don't think it has any checking if it succeed or not.

Reply:
That's a bit troubling... :) In any case, since power management is an important feature in CAM, I think we should make an effort to at least detect if the Install-Idle-Shutdown.sh script fails, regardless of whether it does any error checking internally.

I'm not sure what the appropriate response is but my feeling is that a CAM user would want to know if they deployed an instance that was unable to automatically shut down when idle. Or maybe the appropriate response is to just fail the instance deployment entirely... ?

Reply:
I can create a story and see if we want to do it now?

Reply:
Yeah feels like a backlog item. We've already stretched this story a bit :)

Reply:
I think it's a fairly important one, though, because if the idle shutdown install script fails, it effectively breaks the power management feature. Since we're working on allowing custom images, it seems extra important to me that we make sure our install scripts are robust as we cannot assume they will be executing on a pristine, freshly-pressed OS. :)